simulator/src/window.rs

```diff
@@ -87,10 +135,83 @@ impl Window {
                 } => {
                     return true;
                 }
+                Event::KeyDown {
+                    keycode,
+                    keymod,
+                    repeat,
+                    ..
+                } => {
+                    if let Some(valid_keycode) = keycode {
+                        self.input_events.push(SimulatorEvent::KeyDown {
+                            keycode: valid_keycode,
+                            keymod,
+                            repeat,
+                        });
+                    }
+                }
+                Event::KeyUp {
+                    keycode,
+                    keymod,
+                    repeat,
+                    ..
+                } => {
+                    if let Some(valid_keycode) = keycode {
+                        self.input_events.push(SimulatorEvent::KeyUp {
+                            keycode: valid_keycode,
+                            keymod,
+                            repeat,
+                        });
+                    }
+                }
+                Event::MouseButtonUp {
+                    x, y, mouse_btn, ..
+                } => {
+                    let point = map_input_to_point((x, y), self.scale, self.pixel_spacing);
+                    self.input_events
+                        .push(SimulatorEvent::MouseButtonUp { point, mouse_btn });
+                }
+                Event::MouseButtonDown {
+                    x, y, mouse_btn, ..
+                } => {
+                    let point = map_input_to_point((x, y), self.scale, self.pixel_spacing);
+                    self.input_events
+                        .push(SimulatorEvent::MouseButtonDown { point, mouse_btn });
+                }
+                Event::MouseWheel {
+                    x, y, direction, ..
+                } => {
+                    let scroll_delta = if direction == MouseWheelDirection::Flipped {
+                        Point::new(-1 * x, -1 * y)
+                    } else {
+                        Point::new(x, y)
+                    };
+                    self.input_events.push(SimulatorEvent::MouseWheel {
+                        scroll_delta,
+                        direction,
+                    });
+                }
                 _ => {}
             }
         }
 
         false
     }
+
+    /// Return all captured input events
+    pub fn get_input_events(&mut self) -> Vec<SimulatorEvent> {
+        let input_events = self.input_events.clone();
+        self.clear_event_queue();
```

Comment:
Is the additional method `clear_event_queue` necessary? It's only called once and
```suggestion
        self.input_events.clear();
```
isn't much harder to understand (maybe with an additional comment).

Reply:
D'oh, you're right. That was an oversight on my part.